builder/googlecompute/step_import_os_login_ssh_key.go

```diff
@@ -144,14 +142,14 @@ func (s *StepImportOSLoginSSHKey) Cleanup(state multistep.StateBag) {
 	ui.Message("SSH public key for OSLogin has been deleted!")
 }
 
-func tokeninfo(ctx context.Context, impersonatesa string) (*oauth2.Tokeninfo, error) {
+func tokeninfo(ctx context.Context, config *Config) (*oauth2.Tokeninfo, error) {
+	var err error
 	var opts []option.ClientOption
-
-	opts = append(opts, option.WithUserAgent(useragent.String(version.PluginVersion.FormattedVersion())))
-
-	if impersonatesa != "" {
-		opts = append(opts, option.ImpersonateCredentials(impersonatesa))
+	opts, err = NewClientOptionGoogle(config.account, config.VaultGCPOauthEngine, config.ImpersonateServiceAccount, config.AccessToken)
```

Comment:
Looks like this is missing the `WithUserAgent` call

Reply:
It's already included in the function NewClientOptionGoogle

https://github.com/hashicorp/packer-plugin-googlecompute/blob/fe29ac0dfa7305ef7184ca3fe258b92525961ade/builder/googlecompute/driver_gce.go#L132

Reply:
Cool !